app/assets/javascripts/form-validator.js

```diff
@@ -37,15 +37,23 @@ var FormValidator = function(root) {
 
   this.addInputValidation = function(inputId, validator) {
     var onChange = function (input) {
+      var inputValidationError = root.getElementById(input.id + 'ValidationError');
       if (validator.isValid(input.value)) {
+        inputValidationError.classList.add('display-none');
         input.classList.add('valid');
       } else {
+        if(input.value.length>0){
+          inputValidationError.classList.remove('display-none');
+        }
         input.classList.remove('valid');
+
       }
     };
     var input = document.getElementById(inputId);
     input.addEventListener('change', onChange.bind(this, input));
```

Comment:
Actually I had it like you proposed it already before... so it is ok for me to do it way. Lars and Guilherme made the changes afterwards so that it became that way. I will change it tomorrow on master together with the changes Esther may want to be done.